src/journal.rs

```diff
@@ -325,13 +326,25 @@ impl Journal {
         Ok(cursor)
     }
 
-    /// Returns timestamp at which current journal is recorded
+    /// Returns timestamp at which current journal entry is recorded.
     pub fn timestamp(&self) -> Result<time::SystemTime> {
         let mut timestamp_us: u64 = 0;
         sd_try!(ffi::sd_journal_get_realtime_usec(self.j, &mut timestamp_us));
         Ok(system_time_from_realtime_usec(timestamp_us))
     }
 
+    /// Returns monotonic timestamp and boot ID at which current journal entry is recorded.
+    pub fn monotonic_timestamp(&self) -> Result<(u64, Id128)> {
```

Comment:
It looks like this is missing an input parameter compared to the C libary, i.e. the `If the boot ID parameter is passed NULL` case. Would that better be a separate method or a `current_boot_only: bool` parameter here? (I'm for the latter)

Reply:
Yeah, good point. I made this a separate function instead because otherwise it'd require a separate `sd_id128_get_boot()` call afterwards in the `current_boot_only == true` case to still return the boot ID. Or I guess we could've changed the return type to something like `Result<(u64, Option<Id128>)`, but that just looks painful to use.